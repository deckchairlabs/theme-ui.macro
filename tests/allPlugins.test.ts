import fs from 'fs'
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
  plugin,
  pluginName: 'theme-ui.macro',
  title: 'All Plugins',
  snapshot: false,
  babelOptions: {
    presets: ['@babel/preset-typescript'],
    filename: __filename,
  },
  pluginOptions: {
    themeUI: {
      customProperties: {},
      generateStylesheet: {
        selectors: { buttons: '.button', layout: '.layout' },
        output: path.resolve(__dirname, 'generated/theme-all.css'),
      },
      generateTSDeclaration: {
        output: path.join(__dirname, 'generated/theme-all.d.ts'),
      },
    },
  },
  fixtures: path.join(__dirname, 'plugins/allPlugins'),
})

test('generated stylesheet matches snapshot', () => {
  const content = fs.readFileSync(
    path.resolve(__dirname, 'generated/theme-all.css')
  )
  expect(content.toString()).toMatchSnapshot()
})

test('generated declaration file matches snapshot', () => {
  const content = fs.readFileSync(
    path.resolve(__dirname, 'generated/theme-all.d.ts')
  )
  expect(content.toString()).toMatchSnapshot()
})
