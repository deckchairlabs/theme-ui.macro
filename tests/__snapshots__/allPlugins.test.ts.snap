// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generated declaration file matches snapshot 1`] = `
declare module "@theme-ui/css" {
    export interface Theme {
        styles: {
            root: {
                fontSize: 1;
            };
        };
        colors: {
            primary: "red";
            secondary: "blue";
            black: "#000";
            dark: {
                primary: "white";
            };
            light: {
                primary: "black";
            };
        };
        space: [
            0,
            4,
            8,
            16
        ];
        borderWidths: [
            0,
            1,
            2
        ];
        layout: {
            spacing: {
                large: {
                    margin: 3;
                };
            };
        };
        buttons: {
            base: {
                p: 3;
                paddingX: [
                    2,
                    3
                ];
                borderWidth: 1;
            };
            primary: {
                backgroundColor: "primary";
            };
        };
    }
}
export {};

`;

exports[`generated stylesheet matches snapshot 1`] = `
:root {
    --theme-ui-colors-primary: red;
    --theme-ui-colors-secondary: blue;
    --theme-ui-colors-black: #000;
    --theme-ui-colors-dark-primary: white;
    --theme-ui-colors-light-primary: black;
    --theme-ui-space-0: 0px;
    --theme-ui-space-1: 4px;
    --theme-ui-space-2: 8px;
    --theme-ui-space-3: 16px;
    --theme-ui-borderWidths-0: 0px;
    --theme-ui-borderWidths-1: 1px;
    --theme-ui-borderWidths-2: 2px
}
.button-base {
    padding: var(--theme-ui-space-3);
    padding-left: var(--theme-ui-space-2);
    padding-right: var(--theme-ui-space-2);
    border-width: var(--theme-ui-borderWidths-1);
}
@media screen and (min-width: 40em) {
    .button-base {
        padding-left: var(--theme-ui-space-3);
        padding-right: var(--theme-ui-space-3)
    }
}
.button-primary {
    background-color: var(--theme-ui-colors-primary)
}
.layout-spacing-large {
    margin: var(--theme-ui-space-3)
}
`;
